### PR TITLE
TFX Message exit handler

### PR DESCRIPTION
**Pull request by hanneshapke**
Implementation for  #101. 

- [x] Tests pass #TODO
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

**Review by Udik2021 — APPROVED** [review]

**Comment by rcrowe-google** [comment]
LMK when this is ready to merge

**Review by rcrowe-google — APPROVED** [review]
Thanks, and sorry for the delay!

**Comment by rcrowe-google** [comment]
Please also update the release notes. 😁

**Comment by hanneshapke** [comment]
@rcrowe-google @casassg Can I please get a review and `lgtm` if things are ok?

**Comment by hanneshapke** [comment]
@rcrowe-google @casassg This PR is now fully ready. Test are working again.

**Comment by rcrowe-google** [comment]
Missing files to be approved: /.github/workflows/ci.yml. Potential owners: @casassg and @vulkomilev

File: .github/workflows/ci.yml
Comment (casassg) [inline .github/workflows/ci.yml:34]: why are we removing this btw?

Comment (hanneshapke) [reply .github/workflows/ci.yml:34]: I thought we only support between min and max tfx versions with 1.4.0 as min.

File: tfx_addons/message_exit_handler/proto/slack.proto
Comment (hanneshapke) [reply tfx_addons/message_exit_handler/proto/slack.proto:14]: I am open to pydantic, etc. It just followed the TFX design patterns from the BulkInferrer, InfraValidator, etc. I think the benefit is that the information can be serialized as passed to the component as a string instead of passing every argument individually.

What non-proto structure would you recommend? Plain `json`?

Comment (casassg) [reply tfx_addons/message_exit_handler/proto/slack.proto:14]: pydantic accepts serializing from/to json so yes.

File: .github/workflows/ci.yml
Comment (casassg) [reply .github/workflows/ci.yml:34]: true. Fixed in https://github.com/tensorflow/tfx-addons/pull/119. Seems the file had an typing error

File: tfx_addons/message_exit_handler/proto/slack.proto
Comment (hanneshapke) [reply tfx_addons/message_exit_handler/proto/slack.proto:14]: @casassg Let's turn the feedback into implementation guidelines for the repo. I think those considerations are good to consider.

Comment (hanneshapke) [reply tfx_addons/message_exit_handler/proto/slack.proto]: Agreed. But this is a port from an internal package and we used protos heavily at Digits (a bunch of folks at Digits worked at Google previously). I ported everything  to Pydantic. I think we should collect a few more of those guidelines and add them to the readme.

File: tfx_addons/message_exit_handler/component_test.py
Comment (hanneshapke) [reply tfx_addons/message_exit_handler/component_test.py]: Addressed, thank you for the reference.

File: setup.py
Comment (hanneshapke) [reply setup.py]: Updated.

File: tfx_addons/message_exit_handler/proto/slack.proto
Comment (hanneshapke) [reply tfx_addons/message_exit_handler/proto/slack.proto]: Switched to Pydantic

**Comment by casassg** [comment]
once pre-commit passes it should be merged automatically

**Comment by hanneshapke** [comment]
Sounds good. I'll address the pre-commit issues probably tonight.

**Pull request merged**
